src/server.tsx

```diff
@@ -179,6 +179,30 @@ const buildEpicData = async (
     };
 };
 
+// TODO implement this stub
+const buildBannerData = async (
+    pageTracking: EpicPageTracking,
+    targeting: EpicTargeting,
+    params: Params,
+): Promise<{}> => {
+    //TODO create return types specific to the banner
+    pageTracking && targeting && params; //TODO: remove
+
+    const moduleUrl =
+        process.env.NODE_ENV === 'production'
+            ? 'https://contributions.theguardian.com/banner.js'
+            : 'http://localhost:3040/banner.js';
```

Comment:
I think the pattern you've used here is the one we're using elsewhere, but I wonder if instead of basing the host on the `NODE_ENV` we could just use the host of the _current_ request? (I think in express this comes from `request.headers.host` so we'd have to thread it through to here). I _think_ that's what you'd always want. So this could be simplified to:

```
const moduleUrl = `${currentHost}/banner.js`
```

What do you reckon?

Reply:
Good idea, I think you'd have to be trying something quite unusual for that not to work. Have updated, although I had to use the protocol as well because the host doesn't include that I believe.